rust/lance-core/src/cache.rs

```diff
@@ -114,6 +114,7 @@ impl FileMetadataCache {
 
     pub fn size(&self) -> usize {
         if let Some(cache) = self.cache.as_ref() {
+            cache.sync();
```

Comment:
Is this mainly for testing purposes?  Or are there non-testing reasons we need this to be accurate?  I don't think it's a problem as we _shouldn't_ be calling `size` in a loop so seems fine, just curious.